packages/next-playwright/README.md

```diff
@@ -0,0 +1,84 @@
+# @next/playwright
+
+### Install `@next/playwright` in your project
+
+```sh
+npm install -D @next/playwright
+```
+
+### Update `playwright.config.ts`
+
+```javascript
+import { defineConfig } from '@next/playwright'
+
+export default defineConfig({
+  webServer: {
+    command: 'pnpm dev -- --experimental-test',
+    url: 'http://localhost:3000',
+  },
+})
+```
+
+### Use the `@next/playwright` to create tests
+
+```javascript
+import { test, expect } from '@next/playwright'
+
+test('/product/shoe', async ({ page, next }) => {
+  next.onFetch((request) => {
+    if (request.url === 'http://my-db/product/shoe') {
+      return new Response(
+        JSON.stringify({
+          title: 'A shoe',
+        }),
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      )
+    }
+    return 'abort'
+  })
+
+  await page.goto('/product/shoe')
+
+  await expect(page.locator('body')).toHaveText(/Shoe/)
+})
+```
+
+### Or use the `@next/playwright/msw`
+
+```javascript
+import { test, expect } from '@next/playwright/msw'
+
+test.use({
```

Comment:
Will this be scoped to the current test run? E.g. when running multiple tests concurrently.

Reply:
Yes. In the playwright - a group of tests are executed in a separate worker, which gets its own proxy server. And each test propagates an additional testId. On the `next`'s side I store the proxy server's address (port) and the testId via async local storage and thus all embedded fetches are directed to the right proxy/test.